examples/tokenvm/avalanche-ops.aws.md

```diff
@@ -0,0 +1,249 @@
+
+```bash
+avalancheup-aws default-spec \
+...
+--upload-artifacts-aws-volume-provisioner-local-bin ${AWS_VOLUME_PROVISIONER_BIN_PATH} \
+--upload-artifacts-aws-ip-provisioner-local-bin ${AWS_IP_PROVISIONER_BIN_PATH} \
+--upload-artifacts-avalanche-telemetry-cloudwatch-local-bin ${AVALANCHE_TELEMETRY_CLOUDWATCH_BIN_PATH} \
+--upload-artifacts-avalanched-aws-local-bin ${AVALANCHED_AWS_BIN_PATH} \
+--upload-artifacts-avalanchego-local-bin ${AVALANCHEGO_BIN_PATH} \
+```
+
+It is recommended to specify your own artifacts to avoid flaky github release page downloads.
+
+*NOTE*: The `default-spec` (and `apply`) command only provisions nodes, not custom VMs. The subnet and custom VM installation are done via `install-subnet-chain` sub-commands.
+
+### Step 3
+
+The `default-spec` command itselt does not create any resources, and instead outputs the following `apply` and `delete` commands that you can copy and paste:
+
+```bash
+# first make sure you have access to the AWS credential
+aws sts get-caller-identity
+
+# "avalancheup-aws apply" would not work...
+# An error occurred (ExpiredToken) when calling the GetCallerIdentity operation: The security token included in the request is expired
+
+# "avalancheup-aws apply" will work with the following credential
+# {
+#   "UserId": "...:abc@abc.com",
+#   "Account": "...",
+#   "Arn": "arn:aws:sts::....:assumed-role/name/abc@abc.com"
+# }
+```
+
+```bash
+# run the following to create resources
+vi /home/ubuntu/aops-custom-****-***.yaml
+
+avalancheup-aws apply \
+--spec-file-path /home/ubuntu/aops-custom-****-***.yaml
+
+# run the following to delete resources
+avalancheup-aws delete \
+--delete-cloudwatch-log-group \
+--delete-s3-objects \
+--delete-ebs-volumes \
+--delete-elastic-ips \
+--spec-file-path /home/ubuntu/aops-custom-****-***.yaml
+```
+
+That is, `apply` creates AWS resources, whereas `delete` destroys after testing is done.
+
+*SECURITY*: By default, the SSH and HTTP ports are open to public. Once you complete provisioning the nodes, go to EC2 security group to restrict the inbound rules to your IP.
+
+*NOTE*: In rare cases, you may encounter [aws-sdk-rust#611](https://github.com/awslabs/aws-sdk-rust/issues/611) where AWS SDK call hangs, which blocks node bootstraps. If a node takes too long to start, connect to that instance (e..g, use SSM sesson from your AWS console), and restart the agent with the command `sudo systemctl restart avalanched-aws`.
+
+### Step 4
+
+Now that the network and nodes are up, let's install two subnets, each of which runs its own TokenVM. Note that the above, successful `avalancheup-aws apply` run will output the following commands as an example, which you can customize depending on the VMs:
+
+```bash
+# EXAMPLE: write subnet config
+avalancheup-aws subnet-config \
+--log-level=info \
+--proposer-min-block-delay 250000000 \
+--file-path /tmp/subnet-config.json
```

Comment:
We should use a different path here to avoid conflicting with the ./scripts/run.sh

Reply:
`mkdir -p /tmp/avalanche-ops/*`